src/components/addLocation.js

```diff
@@ -1,79 +1,37 @@
-import React, { useRef, useState, useEffect } from "react";
-import mapboxgl from "mapbox-gl";
+import React, { useState} from "react";
+// import mapboxgl from "mapbox-gl";
 import "react-map-gl-geocoder/dist/mapbox-gl-geocoder.css";
 import "mapbox-gl/dist/mapbox-gl.css";
 import Navbar from './Navbar'
 import axios from 'axios'
 import auth from "./auth";
 import { Snackbar } from "@material-ui/core";
-import MapboxGeocoder from '@mapbox/mapbox-gl-geocoder'
+import {connect} from 'react-redux'
+import PublicMap from './publicMap'
+
+
 
-mapboxgl.accessToken = "pk.eyJ1IjoiYnJpYW5iYW5jcm9mdCIsImEiOiJsVGVnMXFzIn0.7ldhVh3Ppsgv4lCYs65UdA";
-const style = {
-  position: 'absolute',
-  top: "110px",
-  bottom: "2px",
-  width: '100%'
-};
-let marker;
-// let message="";
 const addLocation = (props) => {
 
   const [isSnackbarOpen, setisSnackbarOpen] = useState(false);
   const handleClose = () => {
     setisSnackbarOpen(false)
   };
-  const mapContainer = useRef();
-  const [map, setMap] = useState(null);
-  const [coordinates, setCoordinates] = useState({ lng: "", lat: "" })
   const [message, setMessage] = useState("")
 
-  useEffect(() => {
-    var map = new mapboxgl.Map({
-      container: mapContainer.current,
-      style: 'mapbox://styles/mapbox/streets-v11',
-      center: [-96, 36.5],
-      zoom: 9
-    })
-    map.addControl(new mapboxgl.GeolocateControl({
-      positionOptions: {
-        enableHighAccuracy: true
-      },
-      trackUserLocation: true
-    }));
-    map.addControl(new MapboxGeocoder({
-      accessToken: mapboxgl.accessToken,
-      mapboxgl: mapboxgl
-    }));
-    setMap(map)
-  }, [])
 
-  if (map !== null) {
-    map.on('click', function (event) {
-      if (marker != null) {
-        marker.remove()
-      }
-      marker = new mapboxgl.Marker()
-        .setLngLat(event.lngLat)
-        .addTo(map);
-      setCoordinates({
-        ...coordinates,
-        ...event.lngLat
-      })
-    })
-  }
   function handleAddLocation() {
-    axios.post("http://localhost:3001/locations", coordinates, {
+    let center={lng:props.center[0],lat:props.center[1]}
+
+    axios.post("http://localhost:3001/locations",center , {
```

Comment:
Am I right in assuming that this is a WIP and will be moved out to redux action eventually?

Reply:
Yes, I'm working on it.